trio/_util.py

```diff
@@ -280,7 +281,9 @@ class SomeClass(metaclass=Final):
     - TypeError if a subclass is created
     """
 
-    def __new__(cls, name, bases, cls_namespace):
+    def __new__(
+        cls, name: str, bases: tuple[type, ...], cls_namespace: dict[str, t.Any]
```

Comment:
Personally I dislike `Any` except when totally necessary: can we use `object` here?